MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/MatrixApiActivity.java

```diff
@@ -152,18 +170,19 @@ private void makeMapboxMatrixApiCall(final int markerPositionInList, Point point
       @Override
       public void onResponse(Call<MatrixResponse> call,
                              Response<MatrixResponse> response) {
-        List<Double[]> durationsToAllOfTheLocationsFromTheOrigin = response.body().durations();
-        for (int x = 0; x < durationsToAllOfTheLocationsFromTheOrigin.size(); x++) {
-          String finalConvertedFormattedDistance = String.valueOf(new DecimalFormat("#.##")
-            .format(TurfConversion.convertLength(
-              durationsToAllOfTheLocationsFromTheOrigin.get(markerPositionInList)[x],
-              "meters", "miles")));
-          if (x == markerPositionInList) {
-            matrixLocationList.get(x).setDistanceFromOrigin(finalConvertedFormattedDistance);
-          }
-          if (x != markerPositionInList) {
-            matrixLocationList.get(x).setDistanceFromOrigin(finalConvertedFormattedDistance);
-            matrixApiLocationRecyclerViewAdapter.notifyDataSetChanged();
+        if (response.body() != null) {
+          List<Double[]> durationsToAllOfTheLocationsFromTheOrigin = response.body().durations();
```

Comment:
We are fetching durations, the title of the activity mentions time, but we're converting the data (which I understand is travel time) to distance. I think something's off here.